### PR TITLE
gnomeExtensions.unite-shell: init at 44

### PR DESCRIPTION
**Pull request by rhoriguchi**
###### Motivation for this change

Add support for [GNOME unite-shell](https://github.com/hardpixel/unite-shell)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7551,6 +7551,12 @@
     githubId = 395821;
     name = "rht";
   };
+  rhoriguchi  = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7554]: Please put this into a separate commit with the message ``maintainers: add rhoriguchi``.

```suggestion
  rhoriguchi = {
```

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  buildPhase = ''
+    runHook preBuild
+    glib-compile-schemas --strict --targetdir=${uuid}/schemas/ ${uuid}/schemas
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    ${coreutils}/bin/mkdir -p $out/share/gnome-shell/extensions
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:27]:
```suggestion
    mkdir -p $out/share/gnome-shell/extensions
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108714` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.unite-shell</li>
  </ul>
</details>

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    cp -r ${uuid} $out/share/gnome-shell/extensions
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Unite is a GNOME Shell extension which makes a few layout tweaks to the top panel and removes window decorations to make it look like Ubuntu Unity Shell.";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ rhoriguchi ];
+    homepage = "https://github.com/hardpixel/unite-shell";
+    broken = stdenv.lib.versionAtLeast gnome3.gnome-shell.version "3.32";
```
Comment (sikmir) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:37]:
```suggestion
    broken = versionAtLeast gnome3.gnome-shell.version "3.32";
```

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/gnome-shell/extensions
+    cp -r ${uuid} $out/share/gnome-shell/extensions
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Unite is a GNOME Shell extension which makes a few layout tweaks to the top panel and removes window decorations to make it look like Ubuntu Unity Shell.";
```
Comment (sikmir) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:33]:
```suggestion
    description = "Unite is a GNOME Shell extension which makes a few layout tweaks to the top panel and removes window decorations to make it look like Ubuntu Unity Shell";
```
description should not have a period at the end.

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    license = licenses.gpl3;
```
Comment (sikmir) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:34]: gpl3 is deprecated, use gpl3Only or gpl3Plus if possible

Comment (rhoriguchi) [reply pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:34]: I can't change the license. I will open an issue so the owner might change it.

Comment (sikmir) [reply pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:34]: Sure, it's not about changing the license, it's just about clarification. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

Comment (rhoriguchi) [reply pkgs/desktops/gnome-3/extensions/unite-shell/default.nix]: Thank you for clearing that up. I assumed that it is a different license.

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, gnome3, fetchFromGitHub, xorg, glib, coreutils }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:2]:
```suggestion
{ stdenv, gnome3, fetchFromGitHub, xprop, glib, coreutils }:

stdenv.mkDerivation rec {
```

File: pkgs/desktops/gnome-3/extensions/unite-shell/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    repo = "unite-shell";
+    rev = "v${version}";
+    sha256 = "0nqc1q2yz4xa3fdfx45w6da1wijmdwzhdrch0mqwblgbpjr4fs9g";
+  };
+
+  uuid = "unite@hardpixel.eu";
+
+  nativeBuildInputs = [ glib ];
+
+  buildInputs = [ xorg.xprop ];
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:17]:
```suggestion
  buildInputs = [ xprop ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Delete the ``0.1.6`` file.

**Comment by rhoriguchi** [comment]
Sorry for the sloppiness and thank you for the reviews.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108714` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.unite-shell</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
error: Package ‘gnome-shell-extension-unite-shell-44’ in /home/hotpi/src/nixpkgs2/pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:33 is marked as broken, refusing to evaluate.

**Comment by rhoriguchi** [comment]
> error: Package ‘gnome-shell-extension-unite-shell-44’ in /home/hotpi/src/nixpkgs2/pkgs/desktops/gnome-3/extensions/unite-shell/default.nix:33 is marked as broken, refusing to evaluate.

I finally understand why it's broken I was so confused because it worked in my derivation. The issue is, that I interpreted `versionAtLeast` wrongly, what I wanted was `versionOlder`.

It should work now:

```bash
> nixpkgs-review pr 108714

1 package built:
gnomeExtensions.unite-shell
```

Thank you for the help.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108714` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gnomeExtensions.unite-shell</li>
</ul>
</details>
